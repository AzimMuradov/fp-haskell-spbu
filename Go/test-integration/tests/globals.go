var i = 5;

var a = [10] int {1, 2, 3};

func main() {
  println(i);

  inc();
  println(i);

  inc();
  println(i);

  inc();
  println(i);


  println(a);

  swap(0, 5);
  println(a);
}

func inc() { i++; }

func swap(i int, j int) {
  var temp = a[i];
  a[i] = a[j];
  a[j] = temp;
}
