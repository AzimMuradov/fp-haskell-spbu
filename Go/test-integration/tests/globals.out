5
6
7
8
[1 2 3 0 0 0 0 0 0 0]
[0 2 3 0 0 1 0 0 0 0]
